app/models/piece.rb

```diff
@@ -1,9 +1,76 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  PAWN = 'Pawn'.freeze
+  ROOK = 'Rook'.freeze
+  KNIGHT = 'Knight'.freeze
+  BISHOP = 'Bishop'.freeze
+  QUEEN = 'Queen'.freeze
+  KING = 'King'.freeze
+
+  def obstructed?(destination_x, destination_y)
+    return true if self.invalid?(destination_x, destination_y)
+    return true if self.horizontal_or_vertical_obstruction?(destination_x, destination_y)
+    return true if self.diagonal_obstruction?(destination_x, destination_y)
+    false
+  end
+
+  def horizontal_or_vertical?(destination_x, destination_y)
```

Comment:
All of the methods in this file below `obstructed` can be private.

To do that, write 'private' above `horizontal_or_vertical`

```Ruby
private

def horizontal_or_vertical?(destination_x, destination_y)
....
```

http://rubylearning.com/satishtalim/ruby_access_control.html